src/components/withNavigationFallback.tsx

```diff
@@ -0,0 +1,43 @@
+import React, {ComponentType, RefAttributes, ForwardedRef, ReactElement, forwardRef, useContext, useMemo} from 'react';
+import {NavigationContext} from '@react-navigation/core';
+import {NavigationProp} from '@react-navigation/native';
+import {ParamListBase} from '@react-navigation/routers';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+
+type AddListenerCallback = () => void;
+
+type RemoveListenerCallback = () => void;
```

Comment:
Do we really need these types?

Reply:
@rezkiy37 Otherwise, in `NavigationContextValue`, it will look like

```
type NavigationContextValue = {
    isFocused: () => boolean;
    addListener: () => () => void;
    removeListener: () => () => void;
};

```
As for me, these types make it a little bit more readable.

Reply:
Agree, this perspective also makes sense. So, up to you.